share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorService.java

```diff
@@ -91,6 +93,8 @@ public class ShareCoordinatorService implements ShareCoordinator {
     private final Timer timer;
     private final PartitionWriter writer;
     private final Map<TopicPartition, Long> lastPrunedOffsets;
+    private Supplier<Boolean> isShareGroupConfigEnabled = () -> true;
```

Comment:
nit: `shareGroupConfigEnabledSupplier` would be clearer. The current variable name looks like a boolean, but it's a supplier of a boolean.